backend/api/Areas/Project/Controllers/ReportController.cs

```diff
@@ -0,0 +1,187 @@
+using MapsterMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Pims.Api.Areas.Project.Models.Report;
+using Pims.Api.Areas.Project.Models.Snapshot;
+using Pims.Api.Models;
+using Pims.Api.Policies;
+using Pims.Dal;
+using Pims.Dal.Security;
+using Swashbuckle.AspNetCore.Annotations;
+using Entity = Pims.Dal.Entities;
+
+namespace Pims.Api.Areas.Project.Controllers
+{
+    /// <summary>
+    /// ReportController class, provides endpoints for managing project reports.
+    /// </summary>
+    [Authorize]
+    [ApiController]
+    [ApiVersion("1.0")]
+    [Area("projects")]
+    [Route("v{version:apiVersion}/[area]/reports")]
+    [Route("[area]/reports")]
+    public class ReportController : ControllerBase
+    {
+        #region Variables
+        private readonly IPimsService _pimsService;
+        private readonly IMapper _mapper;
+        private readonly PimsOptions _options;
+        #endregion
+
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a ReportController class, initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="pimsService"></param>
+        /// <param name="mapper"></param>
+        public ReportController(IOptionsMonitor<PimsOptions> options, IPimsService pimsService, IMapper mapper)
+        {
+            _options = options.CurrentValue;
+            _pimsService = pimsService;
+            _mapper = mapper;
+        }
+        #endregion
+
+        #region Endpoints
+        /// <summary>
+        /// Get the project report for the specified 'id'.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpGet("{id:int}")]
+        [HasPermission(Permissions.AdminProjects)]
```

Comment:
FYI, I don't find it helpful to put a comment on every instance of the same issue. You can do this if it helps you to keep track of things in the review of course.

Reply:
I thought about this, but felt that if I was specific it may either imply only this one was incorrect.  I probably should have just made a comment at the top saying change all.  But at the time I was reviewing each function independently.  I'll go back and summarize next time to keep it cleaner.  It does have the benefit of tracking each change though.